### PR TITLE
Fix/price feeder trait

### PR DESCRIPTION
**Pull request by MrPai**
1. add price feeder trait
2. fix liquidation ratio & add issues

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -299,14 +299,13 @@ fn testnet_genesis(
                 (CurrencyId::USDT, 9 * RATE_DECIMAL / 10),
                 (CurrencyId::xDOT, 9 * RATE_DECIMAL / 10),
             ],
-            //FIXME :In fact,"liquidation_threshold" should be higher than "collateral_rate",
-            //but for test, let's make it lower
+            //TODO : please refer, https://github.com/parallel-finance/parallel/issues/46
```
Comment (kaichaosun) [inline node/src/chain_spec.rs:302]:
```suggestion
            //TODO: please refer to https://github.com/parallel-finance/parallel/issues/46
```

File: node/src/service.rs [hunk 2]
```diff
@@ -130,7 +130,7 @@ pub fn new_partial_dev(
     )
     .expect("Insert key should succeed");
 
-    //todo change to mock pool
+    //TODO please refer, https://github.com/parallel-finance/parallel/issues/47
```
Comment (kaichaosun) [inline node/src/service.rs:133]:
```suggestion
    //TODO please refer to https://github.com/parallel-finance/parallel/issues/47
```

File: node/src/service.rs [hunk 3]
```diff
@@ -427,7 +427,7 @@ pub fn new_partial(
...
```
Comment (kaichaosun) [inline node/src/service.rs:430]:
```suggestion
    //TODO please refer to https://github.com/parallel-finance/parallel/issues/47
```

File: pallets/ocw-oracle/src/lib.rs [hunk 4]
```diff
@@ -431,11 +428,17 @@ pub mod module {
...
+}
 
-    impl<T: Config> rt_offchain::storage_lock::BlockNumberProvider for Pallet<T> {
-        type BlockNumber = T::BlockNumber;
-        fn current_block_number() -> Self::BlockNumber {
-            <frame_system::Pallet<T>>::block_number()
-        }
+impl<T: Config> rt_offchain::storage_lock::BlockNumberProvider for Pallet<T> {
+    type BlockNumber = T::BlockNumber;
+    fn current_block_number() -> Self::BlockNumber {
```
Comment (kaichaosun) [inline pallets/ocw-oracle/src/lib.rs:435]: Add an empty line between functions.

File: pallets/loans/src/mock.rs [hunk 5]
```diff
@@ -81,8 +73,12 @@ impl frame_system::Config for Runtime {
     type OnSetCode = ();
 }
 
+pub type AccountId = u128;
 pub type BlockNumber = u64;
 
+pub const ALICE: AccountId = 1;
```
Comment (kaichaosun) [inline pallets/loans/src/mock.rs:79]: There is no need to exact a number to Alice in my opinion.

**Review by kaichaosun — COMMENTED** [review]
Looks good, just a few small concerns.

File: pallets/liquidate/src/lib.rs [hunk 6]
```diff
@@ -95,17 +95,16 @@ pub mod module {
 
     #[pallet::config]
     pub trait Config:
-        frame_system::Config
-        + CreateSignedTransaction<Call<Self>>
-        + pallet_loans::Config
-        + pallet_ocw_oracle::Config
+        frame_system::Config + CreateSignedTransaction<Call<Self>> + pallet_loans::Config
```
Comment (yz89) [inline pallets/liquidate/src/lib.rs:98]: Using `pallet_loans::Config` constraint may cause the same problem as this pr to solve. Maybe we can also add `LiquidateBorrow` trait later on.

File: node/src/chain_spec.rs (hunk 1 above)
Comment (MrPai) [reply node/src/chain_spec.rs:302]: get it

File: pallets/ocw-oracle/src/lib.rs (hunk 4 above)
Comment (MrPai) [reply pallets/ocw-oracle/src/lib.rs:435]: ok, no roblem

File: pallets/loans/src/mock.rs (hunk 5 above)
Comment (MrPai) [reply pallets/loans/src/mock.rs:79]: I agree too, as zhou yang is developing a unit test about pallet loans, I think we can leave it for now.

File: pallets/liquidate/src/lib.rs (hunk 6 above)
Comment (MrPai) [reply pallets/liquidate/src/lib.rs:98]: yes, later we can decouple this too

**Review by yz89 — APPROVED** [review]

**Review by GopherJ — APPROVED** [review]

**Pull request merged**